json-rpc/src/util.rs

```diff
@@ -207,3 +209,22 @@ pub fn script_view_from_script_function(script: &ScriptFunction) -> ScriptView {
         ..Default::default()
     }
 }
+
+pub fn sdk_language_from_user_agent(user_agent: Option<&str>) -> String {
+    let ua: String;
+    match user_agent {
+        Some(user_agent_string) => ua = user_agent_string.to_lowercase(),
+        None => ua = "".to_string(),
+    }
+    if ua.is_empty() {
+        return ua;
+    }
```

Comment:
Couple things here. In rust we prefer to be more explicit about variable naming when possible. so don't say `ua` instead say `user_agent`. You can have "shadowed" variables so you can just redefine `user_agent` with a let statement.  Its also generally not idiomatic rust to declare a variable and then later assign it.

```suggestion
    let user_agent = user_agent.unwrap_or("").to_lowercase();
    if user_agent.is_empty() {
        return user_agent;
    }

Reply:
Of course I would probably rather return "unknown" when user_agent is empty instead of an empty string.